File: WdtcCore/src/main/java/org/wdt/wdtc/auth/accounts/Accounts.java
package org.wdt.wdtc.auth.accounts;

import org.wdt.wdtc.auth.User;
import org.wdt.wdtc.manger.FileManger;
import org.wdt.wdtc.manger.URLManger;
import org.wdt.wdtc.utils.StringUtils;

public class Accounts {
    private final AccountsType type;

    public Accounts() {
        this.type = User.getUser().getType() == AccountsType.Offline ? AccountsType.Offline : AccountsType.Yggdrasil;
    }

    public boolean ifAccountsIsOffline() {
        return type != AccountsType.Yggdrasil;
    }

    public String getJvmCommand() {
        return ifAccountsIsOffline()
                ? StringUtils.STRING_EMPTY
                : StringUtils.appendForString(StringUtils.STRING_SPACE, "-javaagent:", FileManger.getAuthlibInjector(),
                "=", URLManger.getLittleskinApi(), StringUtils.STRING_SPACE, "-Dauthlibinjector.yggdrasil.prefetched=", User.getUser().getAPIBase64());
    }


    public enum AccountsType {
        Offline,
        Yggdrasil
    }

}

File: core/core-api/src/main/kotlin/org/wdt/wdtc/core/openapi/auth/Accounts.kt
package org.wdt.wdtc.core.openapi.auth

import org.wdt.wdtc.core.openapi.manger.authlibInjector
import org.wdt.wdtc.core.openapi.manger.littleskinApiUrl
import org.wdt.wdtc.core.openapi.utils.STRING_EMPTY
import org.wdt.wdtc.core.openapi.utils.STRING_SPACE

class Accounts {
	private val type: AccountsType = preferredUser.type
  
  private val ifAccountsIsOffline = type != AccountsType.YGGDRASIL

  val jvmCommand: String
    get() = if (ifAccountsIsOffline)
      STRING_EMPTY
    else buildString {
      append(STRING_SPACE)
      append("-javaagent:")
      append(authlibInjector.canonicalPath)
      append("=")
      append(littleskinApiUrl)
      append(STRING_SPACE)
      append("-Dauthlibinjector.yggdrasil.prefetched=")
	    append(preferredUser.base64Data)
    }

  enum class AccountsType {
    OFFLINE,
    YGGDRASIL
  }
}
